### PR TITLE
fixes sts rolling update minor issues

### PR DESCRIPTION
**Pull request by f41gh7**

### CODE REVIEW

File: controllers/factory/k8stools/sts.go [hunk 1]
```diff
@@ -113,6 +115,12 @@ func performRollingUpdateOnSts(ctx context.Context, podMustRecreate bool, rclien
 	if sts.Spec.Replicas != nil {
 		neededPodCount = int(*sts.Spec.Replicas)
 	}
+	if err := sortStsPodsByID(podList.Items); err != nil {
+		return fmt.Errorf("cannot sort statefulset pods: %w", err)
+	}
+	for _, p := range podList.Items {
```
Comment (Haleygo) [inline controllers/factory/k8stools/sts.go]: Looks like a test code)

**Comment by Haleygo** [comment]
I get the part that if sts's replicas>10, listing pod result can be misordered. 
But why do we need the list to be ordered? We already set `podManagementPolicy: "Parallel"` and also have https://github.com/VictoriaMetrics/operator/blob/b89ce3b1ecc12568f877e69c4fb9fbc00d63babd/controllers/factory/k8stools/sts.go#L149-L153
that could mess the order.

**Comment by f41gh7** [comment]
> I get the part that if sts's replicas>10, listing pod result can be misordered. But why do we need the list to be ordered? We already set `podManagementPolicy: "Parallel"` and also have
> 
> https://github.com/VictoriaMetrics/operator/blob/b89ce3b1ecc12568f877e69c4fb9fbc00d63babd/controllers/factory/k8stools/sts.go#L149-L153
> 
> that could mess the order.

It shouldn't mess an order, since sorting happens after list requests to the kubernetes. It makes even an order for pod updates. Parallel order cannot be applied for this case, since it's manually managed rollout process.

**Review by Haleygo — APPROVED** [review]
LGTM!

**Pull request merged**